.github/workflows/build-lint-test.yml

```diff
@@ -0,0 +1,31 @@
+name: Build, lint, and test
+
+on: [push]
+
+jobs:
+  build-and-test:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v4
+
+    - name: Set up Node.js
+      uses: actions/setup-node@v4
+      with:
+        node-version-file: 'package.json'
+        cache: 'npm'
+
+    - name: Install dependencies
+      run: npm ci
+
+    - name: Build typescript
+      run: npm run tsc
+
+    - name: Build package
+      run: npm run build
+
+    - name: Run ESLint
+      run: npm run lint
```

Comment:
Is this also the order in which it is run? If so, I would run this before the build as it takes less time to run. I think build should be the last step.